File: app/src/main/java/eu/geekhome/rest/configurable/ConfigurableDto.java
package eu.geekhome.rest.configurable;

import com.geekhome.common.localization.Resource;
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class ConfigurableDto {

    @SerializedName("fields")
    private List<FieldDto> _fields;

    private Resource _descriptionRes;
    @SerializedName("class")
    private String _clazz;

    @SerializedName("addNewRes")
    private Resource _addNewRes;

    @SerializedName("titleRes")
    private Resource _titleRes;

    @SerializedName("iconName")
    private String _iconName;

    @SerializedName("children")
    private List<ConfigurableDto> _children;

    @SerializedName("description")
    private Resource _description;

    public ConfigurableDto(Resource titleRes, Resource descriptionRes, Resource addNewRes, String clazz,
                           List<FieldDto> fields, String iconName, List<ConfigurableDto> children) {
        _titleRes = titleRes;
        _descriptionRes = descriptionRes;
        _clazz = clazz;
        _fields = fields;
        _addNewRes = addNewRes;
        _iconName = iconName;
        _children = children;
    }

    public String getClazz() {
        return _clazz;
    }
}

File: modules/module-data/src/main/java/eu/geekhome/data/configurables/ConfigurableDto.kt
package eu.geekhome.data.configurables

import eu.geekhome.data.fields.FieldDefinitionDto
import eu.geekhome.data.localization.Resource

data class ConfigurableDto(
    val titleRes: Resource,
    val descriptionRes: Resource,
    val clazz: String,
    val parentClazz: String?,
    val type: ConfigurableType,
    val fields: List<FieldDefinitionDto>?,
    val addNewRes: Resource?,
    val editRes: Resource?,
    val iconRaw: String,
    val hasAutomation: Boolean,
    val editableIcon: Boolean,
    val taggable: Boolean,
)